gridcomps/History3G/HistoryGridComp.F90

```diff
@@ -49,12 +49,11 @@ subroutine setServices(gridcomp, rc)
 
       do while (ESMF_HConfigIterLoop(iter, iter_begin, iter_end, rc=status))
          _VERIFY(status)
-
-         collection_name = ESMF_HConfigAsStringMapKey(iter, _RC)
+         collection_name = ESMF_HConfigAsString(iter, _RC)
          child_hconfig = make_child_hconfig(hconfig, collection_name)
          child_name = make_child_name(collection_name, _RC)
          call MAPL_AddChild(gridcomp, child_name, user_setservices(collection_setServices), child_hconfig, _RC)
-         call ESMF_HConfigDestroy(child_hconfig, _RC)
+         !call ESMF_HConfigDestroy(child_hconfig, _RC)
```

Comment:
Still debating myself.      It is a general rule that objects should be destroyed in the same scope (or at the same level) where they are created.   So I'm strongly tempted to have the child create a deep copy for using.

Reply:
@tclune I agree, if you can add that we can restore the destroy.